java/src/gov/usgs/consumerclient/ConsumerClient.java

```diff
@@ -333,9 +346,15 @@ public static void main(String[] args) {
 
 						// nullcheck
 						if (message == null) {
+							// sleep a little while before the next loop
+							Thread.sleep(100);
+
 							continue;
 						}
 						if (message.length() == 0) {
+							// sleep a little while before the next loop
+							Thread.sleep(100);
```

Comment:
The number of 
```
Thread.sleep(100);
```
instances in this function seems awkward.

Reply:
I was trying to sleep on every path out of the loop, be it no data, bad message, or successful write

Reply:
If you want to do stuff on every loop exit, then you should setup a section at the bottom of the loop, or outside of it, that checks a flag and then performs your exit routine.
Old and crude people would use a  `goto`, but you're probably better than that.

Reply:
Sorry I meant loop path, the loop itself is while(true)

Reply:
you could even setup an exception handler for a NoUsefulMessage exception, or some similar craziness, and then use try/catch.